library/alloc/src/alloc.rs

```diff
@@ -342,3 +362,30 @@ pub fn handle_alloc_error(layout: Layout) -> ! {
     }
     unsafe { oom_impl(layout) }
 }
+
+#[cfg(not(any(test, bootstrap)))]
+#[doc(hidden)]
+#[allow(unused_attributes)]
+#[unstable(feature = "alloc_internals", issue = "none")]
+pub mod __default_lib_allocator {
+    use crate::alloc::Layout;
+
+    // called via generated `__rust_alloc_error_handler`
+
+    // if there is no `#[alloc_error_handler]`
+    #[rustc_std_internal_symbol]
+    pub unsafe extern "C" fn __rdl_oom(size: usize, _align: usize) -> ! {
+        panic!("memory allocation of {} bytes failed", size)
+    }
+
+    // if there is a `#[alloc_error_handler]`
+    #[rustc_std_internal_symbol]
+    pub unsafe extern "C" fn __rg_oom(size: usize, align: usize) -> ! {
+        let layout = unsafe { Layout::from_size_align_unchecked(size, align) };
+        extern "Rust" {
+            #[lang = "oom"]
+            fn oom_impl(layout: Layout) -> !;
+        }
+        unsafe { oom_impl(layout) }
```

Comment:
Does this mean that, unlike before, the `#[alloc_error_handler]` is now allowed to unwind? There's no `#[rustc_allocator_nounwind]` any more.

Reply:
Hm, but `__rust_alloc_error_handler` above still has `#[rustc_allocator_nounwind]`.

Reply:
Hm, but `__rust_alloc_error_handler` above still has `#[rustc_allocator_nounwind]`. How is that not UB when it ends up calling `__rdl_oom`?

Reply:
Yeah, I blindly copied `#[rustc_allocator_nounwind]`. Sorry. Remove it, if it is safe.

Reply:
Oops, or the other way round, I might have missed it.